lib/deployment.js

```diff
@@ -1,87 +1,67 @@
 "use strict";
 
+var utils = require("./utils");
+
 module.exports = function deployment(request) {
     return {
         list: function list(cb) {
-            request("/api/deployments/", function deploymentListCallback(err, response, body) {
-                if (err) {
-                    return cb(err);
-                }
-
-                cb(null, JSON.parse(body), response);
-            });
+            request("/api/deployments/", utils.createJsonCallback("listing deployments", cb));
         },
+
         get: function get(id, cb) {
-            request("/api/deployments/" + id, function deploymentGetCallback(err, response, body) {
-                if (err) {
-                    return cb(err);
-                }
+            var url = "/api/deployments/" + encodeURIComponent(id);
+            var action = "getting deployment with id " + id;
 
-                cb(null, JSON.parse(body), response);
-            });
+            request(url, utils.createJsonCallback(action, cb));
         },
+
         del: function del(id, cb) {
-            request.del("/api/deployments/" + id, function deploymentDelCallback(err, response, body) {
-                if (err) {
-                    return cb(err);
-                }
+            var url = "/api/deployments/" + encodeURIComponent(id);
+            var action = "deleting deployment with id " + id;
 
-                cb(null, body, response);
```

Comment:
Ya, I think we'll probably want to consider this PR a breaking change anyway, since it modifies behavior in HTTP error scenarios. Might as well clean up rough edges on the API while we're at it.